lib/logflare_web/controllers/api/access_token_controller.ex

```diff
@@ -0,0 +1,74 @@
+defmodule LogflareWeb.Api.AccessTokenController do
+  use LogflareWeb, :controller
+  use OpenApiSpex.ControllerSpecs
+
+  alias Logflare.Auth
+  alias LogflareWeb.OpenApi.Accepted
+  alias LogflareWeb.OpenApi.Created
+  alias LogflareWeb.OpenApi.List
+  alias LogflareWeb.OpenApi.NotFound
+  alias LogflareWeb.OpenApi.Unauthorized
+  alias LogflareWeb.OpenApiSchemas.AccessToken
+  alias LogflareWeb.Api.FallbackController
+
+  action_fallback(FallbackController)
+
+  tags(["management"])
+
+  operation(:index,
+    summary: "List access tokens",
+    responses: %{200 => List.response(AccessToken)}
+  )
+
+  def index(%{assigns: %{user: user}} = conn, _) do
+    tokens = Auth.list_valid_access_tokens(user) |> Enum.map(&maybe_redact_token/1)
+    json(conn, tokens)
+  end
+
+  operation(:create,
+    summary: "Create source",
+    request_body: AccessToken.params(),
+    responses: %{
+      201 => Created.response(AccessToken),
+      401 => Unauthorized.response(),
+      404 => NotFound.response()
+    }
+  )
+
+  def create(%{assigns: %{user: user}} = conn, params) do
+    scopes_input = Map.get(params, "scopes", "")
+
+    with {:scopes, true} <- {:scopes, "partner" not in String.split(scopes_input)},
+         {:ok, access_token} <- Auth.create_access_token(user, params) do
+      conn
+      |> put_status(201)
+      |> json(access_token)
+    else
+      {:scopes, false} -> {:error, :unauthorized}
+      {:error, _} = err -> err
+    end
```

Comment:
don't really agree that it applies in this case. The "anti-pattern" is around handling ambiguous returns from multiple different functions, where there is possible overlap of error tuples when calling different functions.

In our case, the returns are known and clearly defined given the slimness of the function, and the possible error tuples are also known and handled by the fallback controller.

Also, if we were to follow the recommendation, it would add 2 lines to the stacktrace due to the additional wrapper functions, and make the whole function more verbose than it needs to be.

I have added more explicit tuple matching to make the error boundaries clearer. We can relook at this again in the future, but i think further refactoring has no meaningful difference.